### ISSUE TITLE
About optimization method vs mapper

### ISSUE THREAD

**Issue by wty-ustc** [orpatashnik/StyleCLIP#36]
Thanks for your nice work! I have tested your code, however, I find one issue which seems interesting. When I use your optimization method to manipulate the person with purple hair, the result has hardly changed compared with original image. When I use your mapper, the result seems very nice.
![image](https://user-images.githubusercontent.com/55389940/115838053-da7c1700-a44b-11eb-9c62-6fd894a6fcb6.png)![image](https://user-images.githubusercontent.com/55389940/115838066-de0f9e00-a44b-11eb-90b8-ff1aa27289ff.png)
    original image, optimization, mapper, respectively
So I also tried many other descriptions and images, there also exists the same issue. In my opinion, the optimization method should be better than the encoder approach, so I am very confused about this phenomenon. Do you fine the same issue in your experiment? Can you give me some explaination about this issue? Looking forward to your reply:)

**Comment by orpatashnik** [comment]
Hi @wty-ustc ,

Thanks for your interest!
I agree that the optimization method should allow a wider range of manipulations. From our experience, the main issue with it is that it is not very stable and sometimes it requires tuning of the hyperparameters (`l2_lambda`).
In contrast to the instability of the optimization, we found the mapper much more stable. I think the reason for that is the "knowledge" that the mapper has since it was trained on a large dataset. Unlike the optimization which "searches in the dark", the mapper "knows the right direction".
I hope it gives some intuition 😊

**Comment by Jio0728** [comment]
Hello wty-ustc. May I ask you a question?
I wonder if your loss decreased consistently during training when you tried latent optimization.
In my case, the loss was inconsistent and did not tend to decrease when I tried latent optimization.
